test/e2e/case5_policy_automation_test.go

```diff
@@ -135,7 +136,448 @@ var _ = Describe("Test policy automation", func() {
+					context.TODO(), &replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+			}
+			By("Waiting until the status on each cluster is updated to Compliant")
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				Eventually(func() interface{} {
+					status := replicatedPlc.Object["status"]
+
+					return status.(map[string]interface{})["compliant"]
+				}, 30, 1).Should(Equal(string(policiesv1.Compliant)))
+			}
+
+			By("Should not create any new ansiblejob when Compliant, still two ansiblejobs")
+			Consistently(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 15, 1).Should(Equal(2))
+
+			By("Patching policy to make both clusters back to NonCompliant")
+			opt = metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + testNamespace + "." + case5PolicyName,
+			}
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				replicatedPlc.Object["status"] = &policiesv1.PolicyStatus{
+					ComplianceState: policiesv1.NonCompliant,
+				}
+				_, err := clientHubDynamic.Resource(gvrPolicy).Namespace(replicatedPlc.GetNamespace()).UpdateStatus(
+					context.TODO(), &replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+			}
+			By("Waiting until the status on each cluster is updated to NonCompliant")
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				Eventually(func() interface{} {
+					status := replicatedPlc.Object["status"]
+
+					return status.(map[string]interface{})["compliant"]
+				}, 30, 1).Should(Equal(string(policiesv1.NonCompliant)))
+			}
+
+			By("Should only create the third ansiblejob")
+			Eventually(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).Namespace(testNamespace).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+				_, err = utils.KubectlWithOutput("get", "ansiblejobs", "-n", testNamespace)
+				Expect(err).Should(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 30, 1).Should(Equal(3))
+			Consistently(func() interface{} {
+				ansiblejobList, err := clientHubDynamic.Resource(gvrAnsibleJob).Namespace(testNamespace).List(
+					context.TODO(), metav1.ListOptions{},
+				)
+				Expect(err).To(BeNil())
+				_, err = utils.KubectlWithOutput("get", "ansiblejobs", "-n", testNamespace)
+				Expect(err).Should(BeNil())
+
+				return len(ansiblejobList.Items)
+			}, 30, 1).Should(Equal(3))
+
+			By("Patching policy to make both clusters back to Compliant")
```

Comment:
You can delete from this line to the remainder of the test because you already tested that flipping from non-compliant to compliant doesn't trigger a new job. It's unnecessary to test again.